### ISSUE TITLE
[Proposal] Make `Domain proxy integration tests with orc8r` required check

### ISSUE THREAD

**Issue by sklgromek** [magma/magma#12855]
## Brief Context & scope

We would like to have the `Domain proxy integration tests with orc8r` Check from `dp-workflow` required in CI. 
Changes in orc8r possibly can have an impact on the Domain Proxy component, we would like to avoid merging changes which break compatibility between Orchestrator and Domain Proxy.
The Check was improved and recently is robust and we didn't see flakiness in this job

Link to job definition: https://github.com/magma/magma/blob/7fe7fde1607dd015c4210b2d70de0f8efacd78d8/.github/workflows/dp-workflow.yml#L370

## Goals
Prevent any PRs that break this test from being merged into master.

## Proposal
- Make an announcement in #dev that this test will now be mandatory.
- Wait 3 days for any objections
- Turn on the required check!
- Update documentation on GitHub wiki - move check from "non-blocking" to "blocking" table

**Comment by xbend** [comment]
There **were** occasions where a change was merged into `master` without approval of domain proxy maintainers, where mergers/approvers disregarded to the outcome of CI checks. This introduced severe delays in Domain Proxy delivery, broke master functionalities of Domain Proxy and broke CI for everybody.

One notable PR that disregarded the Domain Proxy CI checks outcome was PR #12564.
A fix #12645 was introduced to correct the mistake in #12564. 
As suggested in #12645, this proposal makes the Domain Proxy integration tests a required job.

Worth mentioning here, is that Domain Proxy integration tests are doing partial Orchestrator deployment within `minikube` - so a full container build and helm deploy.

**Comment by xbend** [comment]
@magma/repo-magma-admin 
Please consider reviewing this proposal and make the appropriate changes to the repo settings.

**Comment by Neudrino** [comment]
Could you please link the respective Github action in the proposal description.

**Comment by sklgromek** [comment]
> Could you please link the respective Github action in the proposal description, please.
@Neudrino I am not sure if it possible. I added a link to the job definition.

**Comment by xbend** [comment]
@magma/repo-magma-admin 
Bump.
Please consider reviewing this proposal and make the appropriate changes to the repo settings.

**Comment by maxhbr** [comment]
And as a proposal this needs to go through TSC. @sklgromek, thank you for bringing it up in the corresponding slack channel.

I think a link to the workflow is: https://github.com/magma/magma/actions/workflows/dp-workflow.yml

**Comment by sklgromek** [comment]
> * [The documentation reads:](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#about-branch-protection-rules)
>   > You can enforce certain workflows or requirements before a collaborator can push changes to a branch in your repository, including merging a pull request into the branch, by creating a branch protection rule.
>   
>   
>   This sounds to me, as if only complete workflows can be made required, not single actions.
>   > We would like to have the Domain proxy integration tests (_ci_integration_tests_orc8r) Check from dp-workflow required in CI.
>  
>   
>   Sounds to me as if only the integration test part should be required?
>   _Possible solution:_
>   To make only the integration test required, the workflows could be split. Making use of composite actions could potentially avoid duplication in this effort.

Hi @Neudrino 
It looks like documentation is not consistent in this regard. I also had some doubt about that and checked it on my own
GitHub project where I have administrator access and in the protected branches configuration section you can select only checks individually, not the whole workflow. 

> * The workflow attempts to use Github secrets, e.g. in
>   https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L389
>   to do some minikube installation.
>   However, those secrets are currently not available from PRs, as the `pull_request` trigger is used.
>   https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L9
>   This should lead to (partly) failing actions on PR. To make the secrets available, the [`pull_request_target` trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) should be used instead.
> 

Thanks! I wasn't aware of that. I will create PR with trigger change to `pull_request_target`

**Comment by maxhbr** [comment]
this was approved in https://wiki.magmacore.org/display/HOME/June+13%2C+2022+Meeting

**Comment by sklgromek** [comment]
> this was approved in https://wiki.magmacore.org/display/HOME/June+13%2C+2022+Meeting

Thank You @maxhbr. When can we expect it to be turned on? Should we do something more?

**Comment by maxhbr** [comment]
this is currently blocked by https://github.com/magma/magma/pull/13019

**Comment by maxhbr** [comment]
It is now an required job:
![2022-06-20_11-25-55](https://user-images.githubusercontent.com/1187050/174570761-8dbb396d-0214-47f2-bd16-bbaeec171575.png)

**Comment by thmsschmitt** [comment]
@maxhbr I think we the same issue here as with the `nms` actions. We can't make a skipped workflow required. https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs

**Comment by maxhbr** [comment]
Ok, I will drop it again from the list, and we can fix it

**Comment by xbend** [comment]
@maxhbr @thmsschmitt 
Hey guys, could you reply to @sklgromek

**Comment by maxhbr** [comment]
Just tried to run it, but it seems to be red in a PR that does not touch dp stuff: https://github.com/magma/magma/runs/7230988721?check_suite_focus=true

Am I missing something?

Screenshot:
![2022-07-07_11:57:46](https://user-images.githubusercontent.com/1187050/177747125-f95eacc6-855d-4135-8bf8-db4b5bf42558.png)

**Comment by xbend** [comment]
`Collect Elasticsearch data` job failed because the `Run Integration tests` job failed - so there is no elasticsearch data to collect.

`Run integraion tests` jobs failed because of:
```
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarnpkg.com/atob/-/atob-2.1.2.tgz: Request failed \"503 Service Unavailable\"".
```

Image building process failed to build containers because docker build failed when trying to fetch packages. This job needs to be restarted, failure is not related to integration tests.

**Comment by maxhbr** [comment]
so it is flaky?

**Comment by xbend** [comment]
In this sense every job which is fetching dependencies from internet sources is flaky. We could, in theory, apply some potential countermeasures like "retry 5 times", however that kind of approach may potentially mask more problems and create false positives.

**Comment by xbend** [comment]
There should be no flakiness in Domain Proxy integration tests anymore.
But to run those tests, we build and setup many orc8r containers inside a minikube cluster, and that setup may fail, for example, because some outside resource is unavailable.

**Comment by Neudrino** [comment]
@maxhbr , @thmsschmitt , @Neudrino were debugging the behaviour and configuration.
The configuration now set is:
![2022-07-07_12_35_01](https://user-images.githubusercontent.com/13189449/177762468-133f28cf-fd8b-4e3d-8e28-7c540566bbf5.png)

Your case was special in terms of that it included a `matrix jobs`, which apparently behaves somewhat differently.

:heavy_check_mark: We believe that we now found a configuration which should provide the desired behaviour.
:arrow_right: We can observe the CI for one day and than close this issue.

**Comment by Neudrino** [comment]
_On a side note:_
Triggering the DP integration tests on any CI change seems to be a overkill and waste of resources to me.
:arrow_right: https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L66
We should probably restrict to the `dp-workflow.yml` here as well. I.e. as for [the other filters](https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L63).

FYI @xbend @sklgromek

**Comment by xbend** [comment]
> _On a side note:_ Triggering the DP integration tests on any CI change seems to be a overkill and waste of resources to me. arrow_right https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L66 We should probably restrict to the `dp-workflow.yml` here as well. I.e. as for [the other filters](https://github.com/magma/magma/blob/master/.github/workflows/dp-workflow.yml#L63).

Agreed. Scheduled to be changed.

**Comment by xbend** [comment]
In terms of the existing issue, we will try to redo the dp-workflow in a way, that is more atomic and remove the matrix syntax.

I will link the PR in this issue since the changes will directly target your suggestions - so would be great to have the PR reviewed also in relation to this particular issue.

**Comment by sklgromek** [comment]
@Neudrino @maxhbr I made changes as you suggested in https://github.com/magma/magma/pull/13240
Could you please review it?

**Comment by sklgromek** [comment]
Hi @Neudrino, @maxhbr as PR https://github.com/magma/magma/pull/13240 with suggested improvements was merged could you please try again to enable job Domain proxy integration tests with orc8r as required check?

**Comment by sklgromek** [comment]
@maxhbr @Neudrino @xbend 
Could we please go further with this?

**Comment by xbend** [comment]
Sadly I have no administrative powers here.
We need actions from the repo admins to re-enable the required check constraint. I guess @maxhbr You can re-enable the requirement since I believe you made the change previously.

**Comment by maxhbr** [comment]
yes, I will be working on that. But need to have someone with PRs queued to test whether it is blocking. Just to immediately check that it is not blocking. I will try to do it now

**Comment by sklgromek** [comment]
@maxhbr, I see that also tests without orc8r were marked as required
![image](https://user-images.githubusercontent.com/1929256/180425243-86ec20c4-b541-4bf0-bf4a-04ff6564d136.png)

We don't want to have them required, moreover, we don't need them anymore and have already PR to remove them,
Could you please remove them from the required checks list?

**Comment by maxhbr** [comment]
fixed
